File: 1653_Minimum_Deletions_to_Make_String_Balanced/solution.py
class Solution:
    def minimumDeletions(self, s: str) -> int:
        res, a, b = len(s), 0, 0
        for c in s:
            a += c == "a"
        for c in s:
            a -= c == "a"
            res = min(res, a + b)
            b += c == "b"
        return res

File: 1653_Minimum_Deletions_to_Make_String_Balanced/solution.c
int minimumDeletions(char* s) {
    int n = strlen(s), a = 0, b = 0;
    for (int i = 0; i < n; i++) a += (s[i] == 'a');
    int result = n;
    for (int i = 0; i < n; i++) {
        a -= (s[i] == 'a');
        result = (result > a + b) ? a + b : result;
        b += (s[i] == 'b');
    }
    return result;
}